elyra/metadata/code_snippet.py

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright 2018-2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from .metadata import Metadata, MetadataManager
+
+from traitlets.config.application import Application
+from jupyter_core.application import base_flags
+from traitlets import Instance, Dict, Unicode, Bool
+from elyra._version import __version__
+
+
+class CodeSnippet(Metadata):
+    namespace = 'code-snippets'
+
+
+class AppUtilMixin(object):
```

Comment:
PR #389 moves this to the `mixins.py` file to eliminate redundancy since this existed in the runtime application as well.